### DISCUSSION TITLE
整合包 (@Phrixus2023) 相关讨论

### DISCUSSION THREAD

**Discussion by zhzLuke96** [lenML/Speech-AI-Forge#65] [分享 | Show and tell]
感谢 @Phrixus2023 提供的整合包：
~~https://pan.baidu.com/s/1ZBv6rC9A2P1OwQ2VPDpdzw?pwd=8xvw~~
~~https://pan.baidu.com/s/1Q1vQV5Gs0VhU5J76dZBK4Q?pwd=d7xu~~
https://pan.baidu.com/s/1vCdlnqNo1z2MfhARHfjyTg?pwd=8nkz

> 出处： https://github.com/lenML/ChatTTS-Forge/issues/60#issuecomment-2183233068

开此贴用于讨论整合包相关的问题

**Comment by louyongjiu** [comment]
@Phrixus2023 对打包方式很感兴趣，方便告知整合包 是用什么工具打包的吗

**Reply by Phrixus2023** [reply]
> @Phrixus2023 对打包方式很感兴趣，方便告知整合包 是用什么工具打包的吗

没有用任何工具打包，就是手动用嵌入式python做的整合包

**Comment by Phrixus2023** [comment]
作者的依赖变了，重新打包了一下ChatTTS-Forge20240624，链接: https://pan.baidu.com/s/1Q1vQV5Gs0VhU5J76dZBK4Q?pwd=d7xu 提取码: d7xu

**Reply by ByeLe** [reply]
大佬，这边有最新版本的整合包吗

**Comment by WangLaoShi** [comment]
其实想问的是，能否告诉大家，你打包的方式或者过程或代码，大家想学习一下。

**Reply by Phrixus2023** [reply]
> 其实想问的是，能否告诉大家，你打包的方式或者过程或代码，大家想学习一下。

下个嵌入式的python，然后用系统的python安装依赖，然后把系统的依赖拷贝到嵌入式里面

**Comment by Phrixus2023** [comment]
作者环境更新了，链接：https://pan.baidu.com/s/1vCdlnqNo1z2MfhARHfjyTg?pwd=8nkz
提取码：8nkz，另外不清楚什么原因调节语速和音量大小那里功能用不了。

**Reply by wangfeng35** [reply]
这个整合包默认是用cpu跑么，怎么改成gpu呢？谢谢。

**Reply by HiraHiraHiraHira** [reply]
> 这个整合包默认是用cpu跑么，怎么改成gpu呢？谢谢。

请问知道如何改成GPU了吗

**Reply by wangfeng35** [reply]
> > 这个整合包默认是用cpu跑么，怎么改成gpu呢？谢谢。
> 
> 请问知道如何改成GPU了吗
卸载整合包里的torch torchvision torchaudio，按照下面的方法重新按照。`pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu121 -f https://mirror.sjtu.edu.cn/pytorch-wheels/torch_stable.html`
https://github.com/lenML/ChatTTS-Forge/blob/main/docs/dependencies.md#5-%E5%AE%89%E8%A3%85-python-%E4%BE%9D%E8%B5%96

**Reply by HiraHiraHiraHira** [reply]
不行唉，安装了，执行run.bat,webui上显示的还是torch: 2.3.1+cpu
Successfully installed torch-2.4.0+cu121 torchaudio-2.4.0+cu121 torchvision-0.19.0+cu121

**Reply by levivz** [reply]
> 作者环境更新了，链接：https://pan.baidu.com/s/1vCdlnqNo1z2MfhARHfjyTg?pwd=8nkz 提取码：8nkz，另外不清楚什么原因调节语速和音量大小那里功能用不了。
rubberband 是一个音频时间拉伸和音高转换的C++库，它提供了一个命令行工具 rubberband-cli，用于处理音频文件。如果您在Python中使用 rubberband 库并遇到了 RuntimeError: Failed to execute rubberband. Please verify that rubberband-cli is installed. 错误，这意味着Python无法找到或执行 rubberband-cli 命令

**Reply by SuperMite233** [reply]
有没有打包好的版本啊，折腾了一晚上一看还是cpu（哭笑）

**Reply by whim2fan** [reply]
自己建个python环境，安装torch torchvision torchaudio,然后安装依赖，最后用自己安装的python启动webui即可使用GPU。

#创建conda虚拟环境
conda create chatTTS-Forge python==3.10
#进入环境
conda activate chatTTS-Forge
#cuda版本根据自己实际版本安装nvidia-smi 可以查看；
pip3 install torch torchvision torchaudio --index-url https://mirror.sjtu.edu.cn/pytorch-wheels/cu121
python -m pip install -r requirements.windows.txt

启动webui
python xxxx/webui.py --language zh-CN --no_half
可修改run.bat
conda activate chatTTS-Forge   && python webui.py  --language zh-CN --no_half

**Comment by suyuan168** [comment]
mac os怎么运行？

**Reply by zhzLuke96** [reply]
这个整合包是windows的，所以对于mac，我建议你使用 docker 部署环境 （很简单，具体可参考readme中docker部分）

或者，自行部署，可以参考 [这个文档](https://github.com/lenML/ChatTTS-Forge/blob/main/docs/dependencies.md) 安装依赖后运行脚本即可
